src/main/kotlin/at/hannibal2/skyhanni/discord/command/ManageCommands.kt

```
@@ -0,0 +1,181 @@
+            .addField("Mention", target.asMention, false)
+            .addField("Moderator", moderator.asMention, false)
+            .addField("Reason", reason, false)
+            .addField("Time", passedSince(SimpleTimeMark.now().toString()), false)
+            .setColor(color)
+            .build()
+    }
+
+    fun handleError(error: Throwable): String {
+        return when (error) {
+            is ErrorResponseException -> {
+                when (error.errorResponse) {
+                    ErrorResponse.UNKNOWN_MEMBER ->
+                        "User not in guild!"
+
+                    ErrorResponse.UNKNOWN_USER ->
+                        "User doesn't exist!"
+
+                    ErrorResponse.CANNOT_SEND_TO_USER ->
+                        "Couldn't DM user!"
+
+                    else -> "Discord API error: ${error.errorResponse}"
+                }
+            }
+
+            else -> "${error.message}"
+        }
+    }
+
+    fun User.sendDM(channel: MessageChannel, action: String, reason: String, color: Color = Color.RED) {
+        openPrivateChannel().queue { dm ->
+            val embed = embed("You were $action!", "**Reason:** $reason", color)
+
+            dm.sendMessageEmbeds(embed).queue(
+                { channel.messageSend("Sent DM to ${name}.") },
+                { error -> channel.messageSend(handleError(error)) }
+            )
+        }
+    }
+}
+
+@Suppress("unused")
+class KickCommand : BaseCommand() {
+    override val name: String = "kick"
+    override val description: String = "Kicks a member from the server."
+    override val options: List<Option> = listOf(
+        Option("user", "The @ or id of the member you want to kick."),
+        Option("reason", "The reason why this member is being kicked.")
+    )
+
+    override fun MessageReceivedEvent.execute(args: List<String>) {
+        val mod = member ?: return reply("Internal error getting moderator.")
```

Instead of showing this, rather throw the error fully. We already catch and print it in chat

---

dont really think there is an error that gets thrown here, I honestly just added this mod 100% isnt null